### PR TITLE
[Geek Camp] Add tracing common util

### LINKED ISSUE #4009: Integrate tracing framework into StarRocks 【Moderate】
#  Requirement Description
Integrate tracing framework into StarRocks, currently open-telemetry seems a good fit: https://github.com/open-telemetry/opentelemetry-cpp
1. add opentelementry thirtparty dependency and build successfully
2. add configs about trace colloctor, and common util/manager classes
3. add some trace for demo purpose, e.g. some stream load related RPCs in BE, should be able to see some trace graphs on tracing tool UI, e.g. jaeger

# Mentor
@decster 

# Registration procedure

**Step 1.** Add a comment when you sign up. The comment can be a brief self-introduction and reason for signing up, including what you do, what you're good at, what you're interested in and your technology stack, and how familiar you are with StarRocks. Example:

> I pick! My name is Johnny. I am from StarRocks and I have been working on database kernel development for 1 year. I am interested in OLAP, good at C++ programming, and have been using StarRocks for 6 months. I chose to solve this issue because I am familiar with execution engine.

**Step 2.** On your WeChat, add the StarRocks community assistant "**_StarRocks-1_**" as a WeChat friend and leave your GitHub ID. The assistant will invite you to a WeChat group based on the issue you picked.

**Step 3.** The mentor will communicate with all the participants who picked this issue and assign the task to the best match. 

# Before you start
1. Read the StarRocks [GitHub workflow](https://github.com/StarRocks/community/blob/main/Contributors/guide/workflow.md) carefully, which provides the configurations of BE and FE and describes how to submit a pull request.
2. To learn more about StarRocks, read [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).

# Pull request requirements

1. After you complete code development, upload the code to the StarRocks code repo as a pull request (PR).

  - PR title format: **[SR GeekCamp] + PR title**
  - Labels ：[Geek Camp 2022/feature]
2. After you submit your PR, comment on the status of your PR submission as requested under the Pinned Issue [[StarRocks geek Camp 2022] ](https://github.com/StarRocks/starrocks/issues/4008)page.

  - Comment format: **"I have finished Issue #" + PR link id**

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4009

### CODE REVIEW

**Comment by stdpain** [comment]
crash
```
76          return this->StartSpan(name, common::KeyValueIterableView<T>(attributes),
(gdb) bt
#0  0x0000000007f6d940 in opentelemetry::v1::trace::Tracer::StartSpan<opentelemetry::v1::nostd::span<std::pair<opentelemetry::v1::nostd::string_view, absl::otel_v1::variant<bool, int, long, unsigned int, double, char const*, opentelemetry::v1::nostd::string_view, opentelemetry::v1::nostd::span<bool const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<int const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<long const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<unsigned int const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<double const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<opentelemetry::v1::nostd::string_view const, 18446744073709551615ul>, unsigned long, opentelemetry::v1::nostd::span<unsigned long const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<unsigned char const, 18446744073709551615ul> > > const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<std::pair<opentelemetry::v1::trace::SpanContext, std::initializer_list<std::pair<opentelemetry::v1::nostd::string_view, absl::otel_v1::variant<bool, int, long, unsigned int, double, char const*, opentelemetry::v1::nostd::string_view, opentelemetry::v1::nostd::span<bool const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<int const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<long const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<unsigned int const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<double const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<opentelemetry::v1::nostd::string_view const, 18446744073709551615ul>, unsigned long, opentelemetry::v1::nostd::span<unsigned long const, 18446744073709551615ul>, opentelemetry::v1::nostd::span<unsigned char const, 18446744073709551615ul> > > > > const, 18446744073709551615ul>, (void*)0, (void*)0> (this=0x0, name=..., attributes=..., links=..., options=...)
    at /var/local/thirdparty/installed/include/opentelemetry/trace/tracer.h:76
#1  0x0000000007f68c21 in opentelemetry::v1::trace::Tracer::StartSpan (this=0x0, name=..., attributes=..., links=..., options=...) at /var/local/thirdparty/installed/include/opentelemetry/trace/tracer.h:138
#2  0x0000000007f68922 in opentelemetry::v1::trace::Tracer::StartSpan (this=0x0, name=<error reading variable: Cannot access memory at address 0xffffffffffffffe0>, options=...)
    at /var/local/thirdparty/installed/include/opentelemetry/trace/tracer.h:48
#3  0x0000000007f64e36 in starrocks::Tracer::start_trace (this=0x607000144470, trace_name=...) at /var/local/starrocks/be/src/common/tracer.cpp:40
```

File: be/test/common/tracer_test.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include <gtest/gtest.h>
+
+#include <memory>
+
+namespace starrocks {
+
+class TracerTest : public testing::Test {};
+
+TEST_F(TracerTest, BasicTest) {
+    std::unique_ptr<Tracer> tracer = std::make_unique<Tracer>();
```
Comment (stdpain) [inline be/test/common/tracer_test.cpp:14]: call init here

Comment (stdpain) [inline be/test/common/tracer_test.cpp:14]: call init here

File: be/src/common/tracer.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ *         auto child = tracer->add_span("child");
+ *         sleepFor(Milliseconds(2));
+ *     }
+ * }
+ * ```
+ *
+ */
+class Tracer {
+public:
+    Tracer() = default;
```
Comment (stdpain) [inline be/src/common/tracer.h:45]: remove default constructor

File: thirdparty/build-thirdparty.sh [hunk 3]
```diff
@@ -851,7 +851,7 @@ build_opentelemetry() {
         -DCMAKE_CXX_STANDARD="17" \
         -DCMAKE_INSTALL_PREFIX=${TP_INSTALL_DIR} \
         -DBUILD_TESTING=OFF -DWITH_EXAMPLES=OFF \
-        -DWITH_STL=ON -DWITH_JAEGER=ON
+        -DWITH_STL=OFF -DWITH_JAEGER=ON
```
Comment (xlwh) [inline thirdparty/build-thirdparty.sh:854]: This looks like a bug! problem description reference: https://github.com/open-telemetry/opentelemetry-cpp/issues/1200

Comment (decster) [reply thirdparty/build-thirdparty.sh:854]: You should open another PR to modify thirdparty build

**Comment by decster** [comment]
run starrocks_be_unittest

**Review by decster — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**